src/components/DocumentVideo/Confirm/index.tsx

```diff
@@ -0,0 +1,192 @@
+import style from './style.scss'
+
+import type { CombinedActions, RootState, DocumentCapture } from '~types/redux'
+import type { ErrorProp, StepComponentDocumentProps } from '~types/routers'
+
+const Confirm: FunctionComponent<StepComponentDocumentProps> = ({
+  nextStep,
+  previousStep,
+  triggerOnError,
+}) => {
+  const [{ token }] = useSdkOptions()
+  const [loading, setLoading] = useState(false)
+  const [previewing, setPreviewing] = useState(false)
+  const [error, setError] = useState<ErrorProp | undefined>(undefined)
+  const { translate } = useLocales()
+
+  const dispatch = useDispatch<Dispatch<CombinedActions>>()
+  const apiUrl = useSelector<RootState, string | undefined>(
+    (state) => state.globals.urls.auth_url
+  )
+  const documentFront = useSelector<RootState, DocumentCapture | undefined>(
+    (state) => state.captures.document_front
+  )
+  const documentBack = useSelector<RootState, DocumentCapture | undefined>(
+    (state) => state.captures.document_back
+  )
+  const documentVideo = useSelector<RootState, DocumentCapture | undefined>(
+    (state) => state.captures.document_video
+  )
+
+  const onUploadDocuments = useCallback(async () => {
+    if (!documentFront) {
+      console.error('Front of document not captured')
+      trackException('Front of document not captured')
+      return
+    }
+
+    if (!documentVideo) {
+      console.error('Document video not captured')
+      trackException('Document video not captured')
+      return
+    }
+
+    setLoading(true)
+
+    const mediaUuids = []
+
+    try {
+      const { media_id: frontMediaUuid } = await uploadBinaryMedia(
+        {
+          file: documentFront.blob,
+          filename: documentFront.filename,
+          sdkMetadata: documentFront.sdkMetadata,
+        },
+        apiUrl,
+        token
+      )
+
+      mediaUuids.push(frontMediaUuid)
+      dispatch(
+        actions.deleteCapture({
+          method: 'document',
+          side: 'front',
+        })
+      )
+
+      if (documentBack) {
+        const { media_id: backMediaUuid } = await uploadBinaryMedia(
+          {
+            file: documentBack.blob,
+            filename: documentBack.filename,
+            sdkMetadata: documentBack.sdkMetadata,
+          },
+          apiUrl,
+          token
+        )
+
+        mediaUuids.push(backMediaUuid)
+        dispatch(
+          actions.deleteCapture({
+            method: 'document',
+            side: 'back',
+          })
+        )
+      }
+
+      const { media_id: videoMediaUuid } = await uploadBinaryMedia(
+        {
+          file: documentVideo.blob,
+          filename: documentVideo.filename,
+          sdkMetadata: documentVideo.sdkMetadata,
+        },
+        apiUrl,
+        token,
+        true // includes file HMAC Auth
+      )
+
+      mediaUuids.push(videoMediaUuid)
+
+      const { uuid: documentUuid } = await createV4Document(
+        mediaUuids,
+        apiUrl,
+        token
+      )
```

Comment:
What happens if the client isn't on API v4?

Reply:
Great question! So far the Doc Platform team has worked to provide a backward compatible workaround for `/checks` creation, so the clients who're using v3 API can still use this feature without having to change anything

Reply:
Nice!